app/Http/Middleware/AdminCheck.php

```diff
@@ -35,7 +36,7 @@ public function handle(Request $request, Closure $next): mixed
 			return $next($request);
 		}
 
-		if (!AccessControl::is_admin()) {
+		if (!Auth::check() || !Gate::check('admin')) {
```

Comment:
Why is the first boolean literal required? The `admin` check should do that anyway.

Even if the first boolean literal is harmless it should always as a user should always be authenticated when the middleware passes this point, it is confusing. If no user was authenticated, then an `UnauthenticatedException` should be thrown not an `UnauthorizedException`. But we are already beyond the point of authentication here.

Hence, if I am not completely mistaken, it should be fine to remove the first literal.

```suggestion
		if (!Gate::check('admin')) {
```

Reply:
Fixed (I don't remember which commit). Can this be resolved ?